### PR TITLE
update py-jupyter-client and new setuptools dependency

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-jupyter-client/package.py [hunk 1]
```diff
@@ -25,3 +26,4 @@ class PyJupyterClient(PythonPackage):
     depends_on('py-traitlets', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyter-client/package.py]: The Python version required has been updated to 2.7+ or 3.5+

File: var/spack/repos/builtin/packages/py-jupyter-client/package.py [hunk 2]
```diff
@@ -25,3 +26,4 @@ class PyJupyterClient(PythonPackage):
...
     depends_on('py-jupyter-core', type=('build', 'run'))
     depends_on('py-zmq@13:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyter-client/package.py]: #13411 will rename this package to `py-pyzmq`, so one of us needs to remember to update this depending on which PR gets merged first. I have no personal preference.

File: var/spack/repos/builtin/packages/py-jupyter-client/package.py [hunk 3]
```diff
@@ -25,3 +26,4 @@ class PyJupyterClient(PythonPackage):
...
+    depends_on('py-setuptools', type='build', when='@5:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyter-client/package.py]: Also missing dependencies on:
```python
        'python-dateutil>=2.1',                                                 
        'tornado>=4.1',    
```

**Comment by adamjstewart** [comment]
Ping @healther, can you rebase now that `py-zmq` has been renamed?

**Comment by healther** [comment]
done

**Comment by healther** [comment]
Travis fail is unrelated?

**Pull request merged**